R/rep_len_linter.R

```diff
@@ -0,0 +1,36 @@
+#' Require usage of rep_len(x, n) over rep(x, length.out = n)
+#'
+#' `rep(x, length.out = n)` calls `rep_len(x, n)` "under the hood". The latter
+#'   is thus more direct and equally readable.
+#'
+#' @examples
+#' # will produce lints
+#' lint(
+#'   text = "rep(1:3, length.out = 10)",
+#'   linters = rep_len_linter()
+#' )
+#'
+#' # okay
+#' lint(
+#'   text = "rep(1:3, each = 2L, length.out = 10L)",
```

Comment:
Shouldn't the okay example instead showcase how to do the same using `rep_len()`?

Reply:
I still don't have a clear structure in my head for what belongs under `# okay` examples. Is it examples paralleling the "bad" examples in the previous section? examples of code using the same functions but in "acceptable" ways?

it seems to get "parallelism" right in general since there's often not a 1:1 correspondence between good/bad sections.

anyway I added an example with `rep_len()`.

Reply:
Yes, thus far we have followed the parallelism pattern where we return the same version of the code in a way that will or will not produce a lint.